packages/flutter_tools/test/src/dap/flutter_adapter_test.dart

```diff
@@ -14,8 +14,77 @@ import 'mocks.dart';
 void main() {
   group('flutter adapter', () {
     setUpAll(() {
-    Cache.flutterRoot = '/fake/flutter';
-  });
+      Cache.flutterRoot = '/fake/flutter';
```

Comment:
Is this test actually running on CI? I just noticed that it's under `flutter_tools/test/src`

Reply:
Hmm, good question! Is `test/src` here not tests for things in `src`? (in [analysis_server](https://github.com/dart-lang/sdk/tree/main/pkg/analysis_server/test/src/computer) it is, so perhaps I assumed incorrectly here).

I can't see which bots are actually running these tests at all - I'll push a change to temporarily break one of these and see if it fails the bot...

Reply:
I realised afterwards that the tests here are all split into shard folders. You were correct that they were not running. I've moved them to `general.shard` and confirmed they are running now (some are failing on Windows due to paths - I'll push a fix shortly).

Reply:
Yeah, the way our tests are run on CI is very confusing. This map defines all the named shards that we define elsewhere, and then mapping to functions that know where the test files are: https://github.com/flutter/flutter/blob/master/dev/bots/test.dart#L199

`package:flutter_tools` probably needs some master config file declaring where all the test files are, that we could then parse and then verify that each file ending in `_test.dart` is in one of those directories.